dotcom-rendering/src/components/DecideContainerByTrails.tsx

```diff
@@ -0,0 +1,281 @@
+				<Card33Media33 trail={trails[0]} showAge={true} />
+			</LI>
+			<LI percentage="33.333%" padSides={true} showDivider={true}>
+				<Card33Media33 trail={trails[1]} showAge={true} />
+			</LI>
+			<LI percentage="33.333%">
+				<UL direction="column" showDivider={true}>
+					<LI padSides={true}>
+						<CardDefault trail={trails[2]} showAge={true} />
+					</LI>
+					<LI padSides={true}>
+						<CardDefault trail={trails[3]} showAge={true} />
+					</LI>
+					<LI padSides={true}>
+						<CardDefault trail={trails[4]} showAge={true} />
+					</LI>
+				</UL>
+			</LI>
+		</UL>
+	);
+};
+
+export const SixCardFast = ({ trails }: { trails: DCRFrontCard[] }) => {
+	if (
+		!trails[0] ||
+		!trails[1] ||
+		!trails[2] ||
+		!trails[3] ||
+		!trails[4] ||
+		!trails[5]
+	)
+		return <></>;
+
+	return (
+		<>
+			<UL direction="row" padBottom={true}>
+				<LI percentage="25%" padSides={true}>
+					<Card25Media25 trail={trails[0]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[1]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[2]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[3]} showAge={true} />
+				</LI>
+			</UL>
+			<UL direction="row">
+				<LI percentage="50%" padSides={true}>
+					<CardDefault trail={trails[4]} showAge={true} />
+				</LI>
+				<LI percentage="50%" padSides={true} showDivider={true}>
+					<CardDefault trail={trails[5]} showAge={true} />
+				</LI>
+			</UL>
+		</>
+	);
+};
+export const SevenCardFast = ({ trails }: { trails: DCRFrontCard[] }) => {
+	if (
+		!trails[0] ||
+		!trails[1] ||
+		!trails[2] ||
+		!trails[3] ||
+		!trails[4] ||
+		!trails[5] ||
+		!trails[6]
+	)
+		return <></>;
+
+	return (
+		<>
+			<UL direction="row" padBottom={true}>
+				<LI percentage="25%" padSides={true}>
+					<Card25Media25 trail={trails[0]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[1]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[2]} showAge={true} />
+				</LI>
+				<LI percentage="25%" padSides={true} showDivider={true}>
+					<Card25Media25 trail={trails[3]} showAge={true} />
+				</LI>
+			</UL>
+			<UL direction="row">
+				<LI percentage="33.333%" padSides={true}>
+					<CardDefault trail={trails[4]} showAge={true} />
+				</LI>
+				<LI percentage="33.333%" padSides={true} showDivider={true}>
+					<CardDefault trail={trails[5]} showAge={true} />
+				</LI>
+				<LI percentage="33.333%" padSides={true} showDivider={true}>
+					<CardDefault trail={trails[6]} showAge={true} />
+				</LI>
+			</UL>
+		</>
+	);
+};
+
+export const EightCardFast = ({ trails }: { trails: DCRFrontCard[] }) => {
```

Comment:
I wonder if we could make `trails` a tuple instead of an array

```suggestion
export const EightCardFast = ({ trails }: { trails:[DCRFrontCard,DCRFrontCard,DCRFrontCard,DCRFrontCard,DCRFrontCard,DCRFrontCard,DCRFrontCard,DCRFrontCard] }) => {
```

Might let us avoid these 2nd checks to make sure the trails array is the length we think it is. Then again maybe this just pushes that complexity up into DecideContainerByTrails

Reply:
Yeah I thought about this as well but it really did just move the complexity around - I think given that this is all fairly well contained in one file it should be okay as is :)